azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -281,81 +277,42 @@ def _init_impl(self):
-            self._msal_client = PublicClientApplication(client_id=self._client_id, authority=self._authority_uri)
-
         try:
-            token = self._msal_client.acquire_token_by_refresh_token(refresh_token, self._scopes)
-        except Exception as ex:
-            raise KustoClientError("Unable to obtain with Az-Cli refresh token. Calling 'az login' may fix this issue.\n" + str(ex))
-
-        return self._valid_token_or_throw(token, "Calling 'az login' may fix this issue.")
-
-    def _get_token_from_cache_impl(self) -> dict:
-        token = None
-        if self._msal_client is not None:
-            account = None
-            if self._username is not None:
-                accounts = self._msal_client.get_accounts(self._username)
-                if len(accounts) > 0:
-                    account = accounts[0]
-
-            token = self._msal_client.acquire_token_silent(scopes=self._scopes, account=account, client_id=self._client_id, authority=self._authority_uri)
-
-        return self._valid_token_or_none(token)
+            if self._az_auth_context is None:
+                self._az_auth_context = AzureCliCredential()
 
-    @staticmethod
-    def _get_azure_cli_auth_token() -> dict:
-        """
-        Try to get the az cli authenticated token
-        :return: refresh token
-        """
-        import os
+            self._az_token = self._az_auth_context.get_token(self._kusto_uri)
+            return {TokenConstants.AZ_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.AZ_ACCESS_TOKEN: self._az_token.token}
+        except Exception as e:
+            raise KustoClientError("Failed to obtain Az Cli token for '{0}'\n{1}".format(self._kusto_uri, e))
 
+    async def _get_token_impl_async(self) -> Optional[dict]:
         try:
-            # this makes it cleaner, but in case azure cli is not present on virtual env,
-            # but cli exists on computer, we can try and manually get the token from the cache
-            from azure.cli.core._profile import Profile
-            from azure.cli.core._session import ACCOUNT
-            from azure.cli.core._environment import get_config_dir
-
-            azure_folder = get_config_dir()
-            ACCOUNT.load(os.path.join(azure_folder, "azureProfile.json"))
-            profile = Profile(storage=ACCOUNT)
-            token_data = profile.get_raw_token()[0][2]
+            if self._az_auth_context_async is None:
+                self._az_auth_context_async = AsyncAzureCliCredential()
 
-            return token_data
+            self._az_token = await self._az_auth_context_async.get_token(self._kusto_uri, self._az_kwargs)
+            return {TokenConstants.AZ_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.AZ_ACCESS_TOKEN: self._az_token.token}
+        except Exception as e:
+            raise KustoClientError(
+                "Failed to obtain Az Cli token for '{0}'.\nPlease be sure to use AzCli version 2.3.0 and above.\n{1}".format(self._kusto_uri, e)

```

Comment:
Is this true just for the async version?

Reply:
What do you mean by _this_?

Reply:
No. Fixing...